pyo3-build-config/src/impl_.rs

```diff
@@ -623,7 +623,7 @@ pub fn cross_compiling(
     let cross_lib_dir = env_var("PYO3_CROSS_LIB_DIR");
     let cross_python_version = env_var("PYO3_CROSS_PYTHON_VERSION");
 
-    let target_triple = format!("{}-{}-{}", target_arch, target_vendor, target_os);
```

Comment:
The problem is that this results in `aarch64-apple-macos` instead of `aarch64-apple-darwin`.

Reply:
Thank you! If tests pass and a `Fixed` CHANGELOG entry added, please feel free to merge this!